### DISCUSSION TITLE
Dose this work with roll-up?

### DISCUSSION THREAD

**Discussion by Errorbot1122** [KiiDii/directory-import#10] [Q&A] [answered]
Dose it?

**Comment by KiiDii** [comment, marked as answer]
Hello! What do you mean? Can you describe your case in more detail?

**Reply by Errorbot1122** [reply]
well, I'm trying to make an npm module, and I want to be able to select all the files in a folder. [Roll-up](https://rollupjs.org/) is just a builder for npm packages (like how [three.js](https://github.com/mrdoob/three.js) has a build folder) I want the same thing in my npm package, and I just want to know if it will work with it. 

P.S: I have not USED roll-up yet as I want to see if this work before I put to much time and effort to make something that is broken 🤷

**Reply by KiiDii** [reply]
Hello! In order for the module to work in the final build, you will need to add the directory to which the `directory-import` will refer. If the bundle consists of only one file and does not include the requested directory, then the module will display an error that the target directory was not found.

Anyway, I haven't worked with rollup.js and you should check it out yourself. But I will be grateful if you try it and leave a review here.

P.S. I tested it with `webpack`. I needed to [use copy-webpack-plugin](https://webpack.js.org/plugins/copy-webpack-plugin/) in order to move the target directory to the final bundle so that the module could find the directory during the launch of the final js file.

P.S.S. You can also try to emulate the `file system` if rollup.js can do it. This will trick the module into thinking that it is finding the target directory. However, I can't tell you how to do it.

**Reply by Errorbot1122** [reply]
https://github.com/ANIname/directory-import/discussions/10#discussioncomment-2350830

> Hello! In order for the module to work in the final build, you will need to add the directory to which the `directory-import` will refer. If the bundle consists of only one file and does not include the requested directory, then the module will display an error that the target directory was not found.

I'll try that, thanks @KiiDii (I'll mark this as awser for now)

**Reply by KiiDii** [reply]
Great. Good luck! 🙂
